src/calendar/calendar.tsx

```diff
@@ -0,0 +1,173 @@
+    const [calendarDate, setCalendarDate] = useState<Dayjs>(dayjs());
+    const [showView, setShowView] = useState<View>("Day");
+    const [selectedStartDate, setSelectedStartDate] = useState<string>(""); // YYYY-MM-DD
+
+    useEffect(() => {
+        if (!value) return;
+
+        setSelectedStartDate(value);
+    }, [value]);
+
+    // =============================================================================
+    // Helper Functions
+    // =============================================================================
+
+    const performOnChangeHandler = (changeValue: string) => {
+        if (onChange) {
+            onChange(changeValue);
+        }
+    };
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+
+    const handleLeftArrowClick = () => {
+        switch (showView) {
+            case "Day":
+                setCalendarDate((date) => date.subtract(1, "month"));
+                break;
+            case "Month":
+                setCalendarDate((date) => date.subtract(1, "year"));
+                break;
+            case "Year":
+                setCalendarDate((date) => date.subtract(10, "year"));
+                break;
+        }
+    };
+
+    const handleRightArrowClick = () => {
+        switch (showView) {
+            case "Day":
+                setCalendarDate((date) => date.add(1, "month"));
+                break;
+            case "Month":
+                setCalendarDate((date) => date.add(1, "year"));
+                break;
+            case "Year":
+                setCalendarDate((date) => date.add(10, "year"));
+                break;
+        }
+    };
+
+    const toggleMonthView = () => {
+        setShowView("Month");
+    };
+
+    const toggleYearView = () => {
+        setShowView((prev) => {
+            if (prev === "Month") return "Day";
+            if (prev === "Year") return "Day";
+
+            return "Year";
+        });
+    };
+
+    const handleDateChange = (value: Dayjs) => {
+        const isOtherMonthSelected = calendarDate.month() !== value.month();
+        const stringValue = value.format("YYYY-MM-DD");
+
+        if (isOtherMonthSelected) setCalendarDate(value);
+
+        setSelectedStartDate(stringValue);
+
+        performOnChangeHandler(stringValue);
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderYearHeader = () => {
+        if (showView === "Year") {
+            const beginDecaded =
+                Math.floor(+calendarDate.format("YYYY") / 10) * 10;
+
+            const endDecaded = dayjs(`${beginDecaded}-01-01`)
+                .add(9, "year")
+                .format("YYYY");
```

Comment:
please help to move this to the `CalendarHelper` and rename `Decaded` to `Decade`

btw I think endDecade can be calculated directly with `beginDecade + 9`